### PR TITLE
feat: Windows.Storage.PathIO

### LINKED ISSUE #4252: Add support for PathIO API
## What would you like to be added:

`Windows.Storage.PathIO`

## Why is this needed:

📁

## For which Platform:

- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [x] Skia
  - [x] WPF
  - [x] GTK (Linux)
  - [x] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #4252

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

Not supported.

## What is the new behavior?

Supported.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UWP/Storage/PathIO.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		{
+			if (!(buffer is UwpBuffer inMemoryBuffer))
+			{
+				throw new NotSupportedException("The current implementation can only write a UwpBuffer");
+			}
+
+			await WriteBytesTaskAsync(file, inMemoryBuffer.Data);
+		}
+
+		private static async Task<Encoding> GetEncodingFromFileAsync(IStorageFile file)
```
Comment (jeromelaban) [inline src/Uno.UWP/Storage/PathIO.cs:131]: Looks like a duplicate of :

https://github.com/unoplatform/uno/blob/511e72d2d43b96911e4901e9de055e03c57bdaa6/src/Uno.UWP/Storage/FileIO.cs#L280

Comment (MartinZikmund) [reply src/Uno.UWP/Storage/PathIO.cs]: Yes, it was a convenience copy paste initially, but I removed those and am using `FileIO` directly under the covers 👍

**Review by davidjohnoliver — APPROVED** [review]

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UWP/Storage/PathIO.cs [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+	/// </summary>
+	public static partial class PathIO
+	{
+		/// <summary>
+		/// Reads the contents of the file at the specified path or Uniform Resource Identifier (URI) and returns text.
+		/// </summary>
+		/// <param name="absolutePath">The path of the file to read.</param>
+		/// <returns>When this method completes successfully, it returns the contents of the file as a text string.</returns>
+		public static IAsyncOperation<string> ReadTextAsync(string absolutePath) =>
+			ReadTextTaskAsync(absolutePath).AsAsyncOperation();
```
Comment (dr1rrb) [inline src/Uno.UWP/Storage/PathIO.cs:24]: You should prefer to use `AsyncOperation.FromTask` which gives you access to the `CancellationToken`

**Pull request merged**